Core-Techs-Git/pdb_requester
# 🎉: RELEASE 1.2.0 (#4)

* REMOVE UNNECESSARY CODE

Change http/https strategies to use request module.

* UPDATE DOCUMENTATION

* WRITE CONFIGURATION SERVICE TEST

Setup test environment with jest.

* TEST REQUESTER SERVICE

* TEST PDB REQUESTER MODULE ENTRY POINT

Update Request service test so it doesn't modify http_proxy environment variable.

* IMPLEMENT NEW CONFIGURATION SETTUP

Make use of protocol, host and path keys in services configuration so that when they're specify the requests are made using protocol://host/path/ as base url and any thing specify in the url/uri are add to the path.

* UPDATE TESTS TO FIT NEW CONFIGURATION SETTUP

* UPDATE DOCUMENTATION

* SETTUP JENKINSFILE FOR CONTINUOUS INTEGRATION

* MAKE MODIFICATIONS TO FIX CI PIPELINE

* TRY CI ERROR NOTIFICATION

* FIX BACK TEST ERROR TO VERIFY CI NOTIFICATIONS

* ADD FIXED EVENT IN POST COMPUTATION

* TRY AGAIN NOTIFICATION FOR REGRESSION

* TRY GET BACK TO NORMAL NOTIFICATION

* CHANGE CONFIGURATION SERVICE AND ERROR MANAGMENT

* UPDATE TESTS SET

* CHANGE TYPESCRIPT MODULE ALIAS PATH RESOLVER

* IGGNORE TSCONFIG INSIDE NPM PUBLISH PACKAGE

* UPDATE DEPENDENCIES

* FIX ASSIGN TO READ ONLY PROPERTY 'PATH' OF CONFIG

* 👷(core): LINT COMMITS MESSAGE (#2)

Format commits message to auto generate changelogs when releasing a new version

BREAKING CHANGE: no more misformed message will be commited.

* 📚(core): AUTOMATE DOCUMENTATION GENERATION

Setup code documentation and release changelog process

* 🎉: RELEASE 1.2.0

Co-authored-by: cstephf <scompaore@adimeo.com>
